src/containers/checkout/checkout-form/delivery/courier/courier.js

```diff
@@ -1,23 +1,274 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
+import { useQuery } from '@apollo/client';
 import { TextField } from '@material-ui/core';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Autocomplete from '@material-ui/lab/Autocomplete';
 import { useStyles } from './courier.styles';
 import { CY_CODE_ERR } from '../../../../../configs';
 import { courierInputLabels } from '../../../../../utils/checkout';
-import { TEXT_FIELD_SIZE, TEXT_FIELD_VARIANT } from '../../../../../const/material-ui';
+import { RESET } from '../../../../../const/checkout';
+import {
+  getUkrPoshtaRegions,
+  getUkrPoshtaDistricts,
+  getUkrPoshtaCities,
+  getUkrPoshtaStreets
+} from '../ukrpost/operations/get-ukrpost-data.queries';
 
-const Courier = ({ language, values, handleChange, touched, errors }) => {
+import { MATERIAL_UI_COLOR, TEXT_FIELD_VARIANT } from '../../../../../const/material-ui';
+import errorOrLoadingHandler from '../../../../../utils/errorOrLoadingHandler';
+import { useIsLoadingOrError } from '../../../../../hooks/useIsLoadingOrError';
+
+const Courier = ({ language, values, handleChange, touched, errors, setFieldValue }) => {
   const styles = useStyles();
   const { t } = useTranslation();
 
+  const {
+    loading: getRegionsLoading,
+    error: getRegionsError,
+    data: { getUkrPoshtaRegions: ukrPoshtaRegions } = []
+  } = useQuery(getUkrPoshtaRegions);
+
+  const {
+    loading: getDistrictsLoading,
+    error: getDistrictsError,
+    data: { getUkrPoshtaDistrictsByRegionId: ukrPoshtaDistricts } = []
+  } = useQuery(getUkrPoshtaDistricts, {
+    variables: { id: values.regionId },
+    skip: !values.regionId
+  });
+
+  const {
+    loading: getCitiesLoading,
+    error: getCitiesError,
+    data: { getUkrPoshtaCitiesByDistrictId: ukrPoshtaCities } = []
+  } = useQuery(getUkrPoshtaCities, {
+    variables: { id: values.districtId },
+    skip: !values.districtId
+  });
+
+  const {
+    loading: getStreetsLoading,
+    error: getStreetsError,
+    data: { getUkrPoshtaStreetsByCityId: ukrPoshtaStreets } = []
+  } = useQuery(getUkrPoshtaStreets, {
+    variables: { id: values.cityId },
+    skip: !values.cityId
+  });
+
+  const { isError } = useIsLoadingOrError([
+    getRegionsError,
+    getDistrictsError,
+    getCitiesError,
+    getStreetsError
+  ]);
+  if (isError) return errorOrLoadingHandler(isError);
   return (
     <div className={styles.courierContainer}>
       <h3 className={styles.courierTitle}>{t('delivery.deliveryAddress')}</h3>
       <div className={styles.inputWrapper}>
+        <Autocomplete
+          name='region'
+          onInputChange={(e, value, reason) => {
+            if (reason !== RESET || (reason === RESET && value)) {
+              setFieldValue('region', value);
+            }
+          }}
+          noOptionsText={t('delivery.noRegion')}
+          onChange={(event, value) => {
```

Comment:
Move all event handlers from template level to component level. And put here only methods that you created.

Reply:
Same to all event handlers.